netmaster/k8snetwork/networkpolicy.go

```diff
@@ -16,25 +16,58 @@ import (
 )
 
 const defaultTenantName = "default"
-const defaultNetworkName = "net"
-const defaultSubnet = "10.1.2.0/24"
-const defaultEpgName = "ingress-group"
-const defaultPolicyName = "ingress-policy"
-const defaultRuleID = "1"
+const defaultNetworkName = "default-net"
+const defaultPolicyPriority = 2
+
+type k8sPodSelector struct {
+	TenantName  string
+	NetworkName string
+	GroupName   string
+}
+
+type k8sPolicyPorts struct {
+	Port     int
+	Protocol string
+}
+
+type k8sIngress struct {
+	IngressRules       []k8sPolicyPorts
+	IngressPodSelector []*k8sPodSelector
+}
+
+type k8sNetworkPolicy struct {
+	PodSelector *k8sPodSelector
+	Ingress     []k8sIngress
+}
 
 type k8sContext struct {
-	k8sClientSet *kubernetes.Clientset
-	contivClient *client.ContivClient
-	isLeader     func() bool
+	k8sClientSet  *kubernetes.Clientset
+	contivClient  *client.ContivClient
+	isLeader      func() bool
+	networkPolicy map[string]k8sNetworkPolicy
 }
 
 var npLog *log.Entry
 
 func (k8sNet *k8sContext) handleK8sEvents() {
+	// wait for leader election
 	for k8sNet.isLeader() != true {
 		time.Sleep(time.Second * 10)
 	}
 
+	// wait till network creation
+	for func() error {
+		if nwList, err := k8sNet.contivClient.NetworkList(); err == nil {
+			if nwList != nil && len(*nwList) > 0 {
```

Comment:
Shouldn't you wait for group creation as well, to get the configured groups before you start creating epg for k8s np.